src/main/java/org/ibp/api/brapi/v2/study/StudyResourceV2Brapi.java

```diff
@@ -0,0 +1,81 @@
+package org.ibp.api.brapi.v2.study;
+
+import com.fasterxml.jackson.annotation.JsonView;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.generationcp.middleware.manager.api.LocationDataManager;
+import org.generationcp.middleware.service.api.BrapiView;
+import org.generationcp.middleware.service.api.location.LocationDetailsDto;
+import org.generationcp.middleware.service.api.location.LocationFilters;
+import org.generationcp.middleware.service.api.study.StudyDetailsDto;
+import org.ibp.api.brapi.v1.common.Metadata;
+import org.ibp.api.brapi.v1.common.Pagination;
+import org.ibp.api.brapi.v1.location.Location;
+import org.ibp.api.brapi.v1.location.LocationMapper;
+import org.ibp.api.brapi.v1.study.StudyDetails;
+import org.ibp.api.brapi.v1.study.StudyDetailsData;
+import org.ibp.api.brapi.v1.study.StudyMapper;
+import org.ibp.api.java.study.StudyService;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Api(value = "BrAPI v2 Study Services")
+@Controller
+public class StudyResourceV2Brapi {
```

Comment:
Hi @clarysabel - wanted to get your feedback on this new v2 call. Ulysis put it in a new resource class: `StudyResourceV2Brapi ` located under "v2" package. The result is it appears as **study-resource-v-2-brapi** in Swagger. Do you think this is okay versus putting this new service under existing `StudyResourceBrapi `(which is under v1 package) so that all versions of the same service are together?

Copying @nahuel-soldevilla , @abatac too as they might have some feedback too

Reply:
I will keep both in different packages. If the class is contained in v2 package, I will remove V2 from the class name and add value to the controller so there is no conflict with the beans creation. Having 2 classes with the same name will end having all v1 and v2 services in the same section in Swagger. I will prefer doing this than mixing everything in the v1 packages in our code. 
I wonder if we want to keep same Swagger documentation for v1 and v2? Some projects just split the API documentation per version (we do not have to solve it now, just thinking how we want to handle it?

Reply:
Thanks for your feedback, @clarysabel. I agree as it allows to keep the code more organized.
please take note of the changes versus swagger state when branch testing @christineguiang , @barbacenasmc

Reply:
As for API documentation, I think it's trickier because we have our own BMS versioning.. (v13, v14, v15, etc) and within these versions, there are also different BrAPI versioning. But I agree it would be ideal to somehow split BrAPI v1 and v2 within our own API documentation when uploading to SwaggerHub. I guess we need to spike/explore how to do that, @mcrimi

Reply:
@darla-leafnode  Major versions should not be an issue because the specification is incremental